cloudant-http/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -251,6 +251,21 @@ public HttpConnection execute() throws IOException {
                 connection.setRequestProperty("Content-type", contentType);
             }
 
+            // We set up the output config before the interceptors to allow the configuration to be
+            // modified. For example an interceptor might change the chunk size by calling
+            // context.connection.getConnection().setChunkedStreamingMode(16384);
+            if (input != null) {
+                connection.setDoOutput(true);
+                if (inputLength != -1) {
+                    // TODO on 1.7 upwards this method takes a long, otherwise int
+                    connection.setFixedLengthStreamingMode((int) this.inputLength);
+                } else {
+                    // TODO some situations where we can't do chunking, like multipart/related
+                    /// https://issues.apache.org/jira/browse/COUCHDB-1403
```

Comment:
The comments just got copied when I moved that block. I've discussed with @tomblench and we can't do the first TODO until 1.7 is the minimum supported level. The second TODO isn't really a TODO it is more an FYI, I will update accordingly.